### PR TITLE
Migrate Guava Predicate and Iterator usage to Java's API Predicate and Iterator usage

### PR DESCRIPTION
**Pull request by GameCubeGBA**
There is no need to rely on Google's implementation of predicates since Java has them.

Sadly, for now, we cannot do the same with Function because the JUNG library depends on Google's version.

### CODE REVIEW

File: Ghidra/Debug/Framework-TraceModeling/src/main/java/ghidra/trace/util/WrappingFunctionIterator.java [hunk 1]
```diff
@@ -32,7 +32,7 @@ public WrappingFunctionIterator(Iterator<? extends Function> it) {
 
 	public <T extends Function> WrappingFunctionIterator(Iterator<T> it,
 			Predicate<? super T> filter) {
-		this.it = Iterators.filter(it, filter);
+		this.it = Iterators.filter(it, filter::test);
```
Comment (astrelsky) [inline Ghidra/Debug/Framework-TraceModeling/src/main/java/ghidra/trace/util/WrappingFunctionIterator.java:35]: This doesn't make sense to me. It is already a predicate so why would you create a lambda to make another predicate using `Predicate::test` which is it's `FunctionalInterface` method. Just forward the existing predicate.

Comment (GameCubeGBA) [reply Ghidra/Debug/Framework-TraceModeling/src/main/java/ghidra/trace/util/WrappingFunctionIterator.java]: Does not compile

File: .gitignore [hunk 2]
```diff
@@ -57,6 +57,7 @@ Release
 # dump files
 *.mdmp
 .vs/
+.vscode/
```
Comment (ryanmkurtz) [inline .gitignore:60]: No need to have this as part of the PR

File: GPL/DMG/src/dmg/java/mobiledevices/dmg/ghidra/GRandomAccessFile.java [hunk 3]
```diff
@@ -17,7 +17,6 @@ public class GRandomAccessFile {
 	private static final byte[] EMPTY = new byte[0];
 	private static final int BUFFER_SIZE = 0x100000;
 
-	private File file;
```
Comment (ryanmkurtz) [inline GPL/DMG/src/dmg/java/mobiledevices/dmg/ghidra/GRandomAccessFile.java:20]: No need to have this as part of the PR

**Review by ryanmkurtz — CHANGES_REQUESTED** [review]

**Comment by GameCubeGBA** [comment]
Addressed!

**Review by ryanmkurtz — APPROVED** [review]

File: Ghidra/Framework/Graph/src/main/java/ghidra/graph/graphs/FilteringVisualGraph.java [hunk 4]
```diff
@@ -118,7 +118,7 @@ public Iterator<V> getFilteredVertices() {
 		Predicate<? super V> isFiltered = v -> {
 			return !containsVertex(v) && completeGraph.containsVertex(v);
 		};
-		return Iterators.filter(getAllVertices(), isFiltered);
+		return Iterators.filter(getAllVertices(), isFiltered::test);
```
Comment (ryanmkurtz) [inline Ghidra/Framework/Graph/src/main/java/ghidra/graph/graphs/FilteringVisualGraph.java:121]: There shouldn't be a need to use Guava's `Iterators.filter()` method anymore.  It's a little confusing too.  You should be able to use Java Streams.

**Review by ryanmkurtz — CHANGES_REQUESTED** [review]

**Comment by GameCubeGBA** [comment]
Fixed @ryanmkurtz

**Comment by ryanmkurtz** [comment]
Looks like there are now conflicts with this, and I think there will be more coming.  We will handle the removing of Guava usage internally.

**Pull request closed without merging**